src/packets/base/Struct.ts

```diff
@@ -69,30 +69,107 @@ export class Struct implements Receivable {
     }
 
     const propertyNames = this.getValidPropertyNames();
-    propertyNames.forEach((propertyName, i) => {
+    let i = 0;
+    propertyNames.forEach((propertyName) => {
       const value = data[i];
+      if (i >= data.length) {
+        return;
+      }
+
+      const propertyType = typeof this[propertyName as keyof this];
 
       if (value === undefined) {
+        i++;
         return;
       }
 
       if (propertyName === 'Size') {
         (this[propertyName as keyof this] as unknown as number) =
           (value as number) * this.SIZE_MULTIPLIER;
+        i++;
         return;
       }
 
       if (isArray(value) && value.length === 2) {
         this[propertyName as keyof this] = value[1] as this[keyof this];
         this._raw[propertyName as keyof RawProperties<this>] = value[0];
+        i++;
+        return;
+      }
+
+      if (isArray(value) && isArray(this[propertyName as keyof this])) {
+        this[propertyName as keyof this] = value as this[keyof this];
+        i++;
+        return;
+      }
+
+      if (propertyType == 'object') {
+        i = this.ParseObject(
+          this[propertyName as keyof this],
+          data,
+          i,
+          propertyName,
+        );
         return;
       }
 
       this[propertyName as keyof this] = value as this[keyof this];
+      i++;
     });
 
     log('Data unpacked:', this);
 
     return this;
   }
+
+  public ParseArray(
```

Comment:
Just a naming convention - function names should be in `camelCase`, so `parseArray` and `parseObject`.

Reply:
changed.